languages/elixir/exercises/concept/exceptions/.docs/introduction.md

```diff
@@ -0,0 +1,52 @@
+All errors in Elixir implement the _Exception Behaviour_. Just like the _Access Behaviour_, the _Exception Behavior_ defines callback functions that a module must implement to fulfill the software contract of the behaviour. Once an errors is defined, it has the following properties:
+
+- The module's name defines the error's name.
+- The module defines an error-struct.
+- The module will have a `:message` field.
+- The module can be be used with `raise/1` and `raise/2` to raise the intended error
+
+The _Exception Behaviour_ also specifies two required callbacks: `message/1` and `exception/1`. Despite being required, `message/1` is often left to be implemented by the _behavior_, but it is common to implement `exception/1` instead to get custom error messages.
+
+## Defining an exception
+
+To define an exception from an error module, we use the `defexception` macro function:
+
+```elixir
+# Defines a minimal error, with the name `MyError`
+defmodule MyError do
+  defexception [:message]
+end
+
+# Defines an error with a customized exception/1 function
+defmodule MyCustomizedError do
+  defexception [:message]
+
+  @impl true
+  def exception(value) do
+    case value do
+      [] ->
+        msg = "service did not get what was expected"
+        %MyCustomizedError{message: msg}
+      _ ->
+        %MyCustomizedError{message: value}
```

Comment:
This example shows how to set a default error message in case none was passed when raising an error, correct? Are there any other good examples of use cases for redefining `exception/1`? Because the default message is easier to achieve by setting the default value when creating the struct: `defexception [message: "service did not get what was expected"]`

Reply:
There is some discrepancy here for sure (maybe we'll get to open a PR in the elixir docs).

I made a bad assumption that as I was working from the [`defexception` documentation](https://hexdocs.pm/elixir/Kernel.html?#defexception/1) and  _`Exception Behaviour`_ documentation that it would be internally consistent with the getting started guide.  Which, now on further review. is not.

The [getting started guide](https://elixir-lang.org/getting-started/try-catch-and-rescue.html#errors) uses this example:

```shell
iex> defmodule MyError do
iex>   defexception message: "default message"
iex> end
iex> raise MyError
** (MyError) default message
iex> raise MyError, message: "custom message"
** (MyError) custom message
```

It seems from your comment, @angelikatyborska, that this is the normative way to implement this in your practice.

It would be useful to hear from @NobbZ or @devonestes in regards to this and if there is a practice to avoid, or if we should implement and display them both.

Reply:
Ok, I spoke to @NobbZ in slack about this, some take aways:
- the documentation is very old on this (gitblame puts it at ~6years old) so it may be outmoded
- message/1 is to make a human readable version of the exception struct
- exception/1 is to add additional formatting to the error message based on the value passed to the function

I think the way forward would be to have an example that follows the getting started guide, then have an example that implements the exception/1 callback.